src/cli.rs

```diff
@@ -152,6 +154,23 @@ pub struct RootOpts {
     #[arg(short, long, env = "VECTOR_WATCH_CONFIG")]
     pub watch_config: bool,
 
+    /// Method for watching config
+    ///
+    /// By default `vector` use `inotify` which is recommended for linux based system
+    /// `poll` watcher can be used where inotify not work. ie, attaching config using NFS
+    #[arg(long, default_value = "inotify", env = "VECTOR_WATCH_METHOD")]
```

Comment:
the `ConfigWatcher::Inotify` variant does not always set the `inotify` watcher, it sets the `RecommendedWatcher`, which would be `inotify` on supported platforms and polling in others.

I think we should rename those variants (or at least, the `Inotify` one), as it is a bit misleading for the user (also, teh default value here should be change accordingly)